fingertip/plugins/cleanup.py

```diff
@@ -100,6 +101,49 @@ def tempfiles(older_than='4h', location=None):
                                       temp.AUTOREMOVE_PREFIX not in f))
 
 
+def saviour(expired_for='10d'):
+    if expired_for != 'all':
+        adjusted_time = time.time() - units.parse_time_interval(expired_for)
+    else:
+        adjusted_time = 0
+    root, dirs, files = next(os.walk(path.SAVIOUR + '/_'))
+    for d in (os.path.join(root, x) for x in dirs):
+        lock_path = os.path.join(root, '.' + os.path.basename(d) + '-lock')
+        lock = fasteners.process_lock.InterProcessLock(lock_path)
+        lock.acquire()
+        time_file_path = os.path.join(d, ".last_mirrored")
+        data_path = os.path.join(d, "data")
+        if os.path.exists(time_file_path):
+            remove = _time(time_file_path) <= adjusted_time
+        elif os.path.exists(data_path):
+            remove = _time(data_path) <= adjusted_time
+        else:
+            remove = True
+        if (expired_for == 'all' or remove):
+            assert os.path.realpath(d).startswith(
+                os.path.realpath(path.SAVIOUR)
+            )
+            log.info(f'removing {os.path.realpath(d)}')
+            if not os.path.islink(d):
+                shutil.rmtree(d)
+            else:
+                os.unlink(d)
+        else:
+            log.debug(f'keeping {os.path.realpath(d)}')
+        os.unlink(lock_path)
+        lock.release()
+    _, top_dirs, __ = next(os.walk(path.SAVIOUR))
+    for top_dir in top_dirs:
+        if top_dir == "_" or top_dir.startswith('.'):
+            continue
+        for root, dirs, files in os.walk(os.path.join(path.SAVIOUR, top_dir)):
+            for f in (os.path.join(root, x) for x in files):
+                if not os.path.exists(os.readlink(f)):
+                    os.unlink(f)
+                    log.info(f'removing {f}')
+                else:
+                    log.debug(f'keeping {f}')
+
 def periodic():
```

Comment:
I guess this one should have the saviour cleanup invoked as well, and same for `everything()`.